value/valueunstructured.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package value
+
+import (
+	"fmt"
+	"sync"
+)
+
+var viPool = sync.Pool{
+	New: func() interface{} {
+		return &valueUnstructured{}
+	},
+}
+
+// NewValueInterface creates a Value backed by an "interface{}" type,
+// typically an unstructured object in Kubernetes world.
+func NewValueInterface(v interface{}) Value {
+	vi := viPool.Get().(*valueUnstructured)
+	vi.Value = v
+	return Value(vi)
+}
+
+type valueUnstructured struct {
+	Value interface{}
+}
+
+func (v valueUnstructured) IsMap() bool {
```

Comment:
Is it possible for IsMap and IsNull to return true at the same time?

Reply:
I don't think so